Jellyfin.Server/Migrations/Routines/FixDates.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Database.Implementations;
+using Jellyfin.Server.ServerSetupApp;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Migrations.Routines;
+
+/// <summary>
+/// Migration to fix dates saved in the database to always be UTC.
+/// </summary>
+[JellyfinMigration("2025-06-20T18:00:00", nameof(FixDates))]
+public class FixDates : IAsyncMigrationRoutine
+{
+    private const int PageSize = 5000;
+
+    private readonly ILogger _logger;
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FixDates"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="startupLogger">The startup logger for Startup UI integration.</param>
+    /// <param name="dbProvider">Instance of the <see cref="IDbContextFactory{JellyfinDbContext}"/> interface.</param>
+    public FixDates(
+        ILogger<MoveExtractedFiles> logger,
+        IStartupLogger<MoveExtractedFiles> startupLogger,
+        IDbContextFactory<JellyfinDbContext> dbProvider)
+    {
+        _logger = startupLogger.With(logger);
+        _dbProvider = dbProvider;
+    }
+
+    /// <inheritdoc />
+    public async Task PerformAsync(CancellationToken cancellationToken)
+    {
+        using var context = _dbProvider.CreateDbContext();
+        var sw = Stopwatch.StartNew();
+
+        await FixBaseItemsAsync(context, sw, cancellationToken).ConfigureAwait(false);
+        sw.Reset();
+        await FixChaptersAsync(context, sw, cancellationToken).ConfigureAwait(false);
+        sw.Reset();
+        await FixBaseItemImageInfos(context, sw, cancellationToken).ConfigureAwait(false);
+    }
+
+    private async Task FixBaseItemImageInfos(JellyfinDbContext context, Stopwatch sw, CancellationToken cancellationToken)
+    {
+        int offset = 0;
+
+        var baseQuery = context.BaseItemImageInfos;
+        var records = baseQuery.Count();
+        _logger.LogInformation("Modifying dates for {Count} BaseItemImageInfos.", records);
+
+        var transaction = await context.Database.BeginTransactionAsync(cancellationToken).ConfigureAwait(false);
+        await using (transaction.ConfigureAwait(false))
```

Comment:
No need for a dedicated transaction, those modifications are all within the same SaveChanges which itself uses a transaction